process/asyncExecution/queue/blocksQueue.go

```diff
@@ -0,0 +1,97 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-go/common"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("process/asyncExecution/queue")
+
+// blocksQueue represents a thread-safe queue for storing and processing blockchain headers.
+// It provides methods to add headers at the end or beginning of the queue and retrieve them
+// for processing in a FIFO (First In, First Out) manner
+type blocksQueue struct {
+	mutex           *sync.Mutex
+	cond            *sync.Cond
+	headerBodyPairs []HeaderBodyPair
+	closed          bool
+}
+
+// NewBlocksQueue creates and returns a new instance of blocksQueue
+func NewBlocksQueue() (*blocksQueue, error) {
+	mutex := &sync.Mutex{}
+
+	return &blocksQueue{
+		mutex:           mutex,
```

Comment:
you don't need to keep a reference to the mutex separately as it is available through the cond (`cond.L.Lock()`)